src/Core/System/Currency/CurrencyFormatter.php

```diff
@@ -31,21 +31,20 @@ public function formatCurrencyByLanguage(float $price, string $currency, string
     {
         $decimals ??= $context->getRounding()->getDecimals();
 
-        $locale = $this->languageLocaleProvider->getLocaleForLanguageId($languageId);
-        $formatter = $this->getFormatter($locale, \NumberFormatter::CURRENCY);
+        $formatter = $this->getFormatter(
+            $this->languageLocaleProvider->getLocaleForLanguageId($languageId)
+        );
         $formatter->setAttribute(\NumberFormatter::FRACTION_DIGITS, $decimals);
 
         return (string) $formatter->formatCurrency($price, $currency);
     }
 
-    private function getFormatter(string $locale, int $format): \NumberFormatter
+    private function getFormatter(string $locale): \NumberFormatter
     {
-        $hash = md5(json_encode([$locale, $format], \JSON_THROW_ON_ERROR));
-
-        if (isset($this->formatter[$hash])) {
-            return $this->formatter[$hash];
+        if (isset($this->formatter[$locale])) {
```

Comment:
Can you add a `reset` method and reset this state using `kernel.reset` tag?

Reply:
In the same PR/commit? :-) (because it is actually not related to this change)

Reply:
I added it as a new commit to this PR :-)